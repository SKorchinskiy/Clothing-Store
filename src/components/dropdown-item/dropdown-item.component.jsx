import { memo } from "react";

import {
  DropdownItemDetails,
  DropdownItemContainer,
} from "./dropdown-item.styles";

function DropdownItem({ cartItem }) {
  return (
    <DropdownItemContainer key={cartItem.id}>
      <img src={cartItem.imageUrl} alt={cartItem.name} />
      <DropdownItemDetails>
        <div>{cartItem.name}</div>
        <div>
          {cartItem.quantity} x ${cartItem.price}
        </div>
      </DropdownItemDetails>
    </DropdownItemContainer>
  );
}

export default memo(DropdownItem);
