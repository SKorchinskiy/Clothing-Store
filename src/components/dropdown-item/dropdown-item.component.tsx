import { memo } from "react";

import {
  DropdownItemDetails,
  DropdownItemContainer,
} from "./dropdown-item.styles";

import { CartItemType } from "../cart-item/cart-item.component";

type DropdownItemProps = {
  cartItem: CartItemType;
};

function DropdownItem({ cartItem }: DropdownItemProps) {
  return (
    <DropdownItemContainer key={cartItem.id}>
      <img src={cartItem.imageUrl} alt={cartItem.name} />
      <DropdownItemDetails>
        <div>{cartItem.name}</div>
        <div>
          {cartItem.quantity} x ${cartItem.price}
        </div>
      </DropdownItemDetails>
    </DropdownItemContainer>
  );
}

export default memo(DropdownItem);
