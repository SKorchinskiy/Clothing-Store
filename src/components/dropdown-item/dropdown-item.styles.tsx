import styled from "styled-components";

export const DropdownItemDetails = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-left: 10px;
`;

export const DropdownItemContainer = styled.div`
  display: flex;
  padding: 20px;
`;
