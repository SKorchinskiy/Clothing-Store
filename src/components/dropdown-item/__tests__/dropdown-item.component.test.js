import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

import DropdownItem from "../dropdown-item.component";

import { dropdownItemStub } from "./stubs/dropdown-item.stub";

describe("<DropdownItem /> component", () => {
  it("should render initial UI properly", () => {
    const { asFragment } = render(<DropdownItem cartItem={dropdownItemStub} />);

    const imageElement = screen.getByAltText(dropdownItemStub.name);
    expect(imageElement).toBeInTheDocument();

    const nameElement = screen.getByText(dropdownItemStub.name);
    expect(nameElement).toBeInTheDocument();

    const detailsElement = screen.getByText(
      `${dropdownItemStub.quantity} x $${dropdownItemStub.price}`
    );
    expect(detailsElement).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });
});
