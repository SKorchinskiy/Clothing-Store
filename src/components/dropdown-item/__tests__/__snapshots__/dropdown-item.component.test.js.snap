// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownItem /> component should render initial UI properly 1`] = `
<DocumentFragment>
  <div
    class="sc-grXZZQ duBxGz"
  >
    <img
      alt="dropdown/item/1"
      src="dropdown/item/imageUrl/1"
    />
    <div
      class="sc-bcPKhP kovsxx"
    >
      <div>
        dropdown/item/1
      </div>
      <div>
        1 x $10
      </div>
    </div>
  </div>
</DocumentFragment>
`;
